src/nimib/sources.nim

```diff
@@ -0,0 +1,202 @@
+import std/[macros, sugar]
+import std/[
+  parseutils, 
+  strutils
+  ]
+import types
+
+
+
+# Credits to @haxscramper for sharing his code on reading the line info
+# And credits to @Yardanico for making a previous attempt which I have taken much inspiration from.
+
+type
+  Pos* = object
+    line*: int
+    column*: int
+
+proc toPos*(info: LineInfo): Pos =
+  Pos(line: info.line, column: info.column)
+
+proc startPos*(node: NimNode): Pos =
+  ## Has column info
```

Comment:
this (and similar in finishPos) look more like normal comment than function documentation, correct?

Reply:
Yep, it's still just a copy paste of hax's code so will go through and maybe add a few documentation comments as well.